subprojects/resources-gcs/src/integTest/groovy/org/gradle/integtests/resource/gcs/fixtures/GcsServer.groovy

```diff
@@ -0,0 +1,497 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resource.gcs.fixtures
+
+import groovy.xml.StreamingMarkupBuilder
+import org.gradle.integtests.resource.gcs.fixtures.stub.HttpStub
+import org.gradle.integtests.resource.gcs.fixtures.stub.StubRequest
+import org.gradle.test.fixtures.file.TestDirectoryProvider
+import org.gradle.test.fixtures.server.RepositoryServer
+import org.gradle.test.fixtures.server.http.HttpServer
+import org.joda.time.DateTimeZone
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.format.DateTimeFormatter
+import org.joda.time.tz.FixedDateTimeZone
+import org.mortbay.jetty.Request
+import org.mortbay.jetty.handler.AbstractHandler
+
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+import java.security.MessageDigest
+
+class GcsServer extends HttpServer implements RepositoryServer {
+
+    public static final String BUCKET_NAME = "tests3bucket"
+    private static final DateTimeZone GMT = new FixedDateTimeZone("GMT", "GMT", 0, 0)
+    protected static final DateTimeFormatter RCF_822_DATE_FORMAT = DateTimeFormat.forPattern("EEE, dd MMM yyyy HH:mm:ss z")
+            .withLocale(Locale.US)
+            .withZone(GMT);
+
```

Comment:
We can drop the X_AMZ stuff for google, and the Server header we could drop or make whatever googles header is. It doesnt hurt to have extra eithers I guess but if we dont need them for GCS might as well drop em